core/arch/arm/pta/system.c

```diff
@@ -600,6 +601,176 @@ static TEE_Result system_remap(struct tee_ta_session *s, uint32_t param_types,
 	return res;
 }
 
+/* ldelf has the same architecture/register width as the kernel */
+#ifdef ARM32
+static const bool is_arm32 = true;
+#else
+static const bool is_arm32;
+#endif
+
+static TEE_Result call_ldelf_dlopen(struct user_ta_ctx *utc, TEE_UUID *uuid,
+				    uint32_t flags)
+{
+	uaddr_t usr_stack = utc->ldelf_stack_ptr;
+	TEE_Result res = TEE_ERROR_GENERIC;
+	struct dl_entry_arg *arg = NULL;
+	uint32_t panic_code = 0;
+	uint32_t panicked = 0;
+
+	usr_stack -= ROUNDUP(sizeof(*arg), STACK_ALIGNMENT);
+	arg = (struct dl_entry_arg *)usr_stack;
+
+        res = tee_mmu_check_access_rights(utc, TEE_MEMORY_ACCESS_READ |
+                                          TEE_MEMORY_ACCESS_ANY_OWNER,
```

Comment:
Copy/paste again :(
Right, should this read `TEE_MEMORY_ACCESS_READ | TEE_MEMORY_ACCESS_WRITE`? The user stack is obviously writable by the TA.

Reply:
If I remove `TEE_MEMORY_ACCESS_ANY_OWNER` the test fails... Any idea why? @jenswi-linaro is there something special with the `ldelf` stack?

Reply:
I guess it's the `VM_FLAGS_NONPRIV` bits in the region mapping the `ldelf` stack. That flag is used to indicate if a memory map is considered TA private or not.

Early on I was thinking that `ldelf` and its mappings should be removed once `ldelf` was done. Since `ldelf` now is always there as a helper we could consider disabling it instead in order to guarantee its integrity.

Reply:
You mean clearing the `VM_FLAG_LDELF` flag? (sounds a bit counter-intuitive to have ldelf stuff **not** flagged like this).

Reply:
No, `TEE_MEMORY_ACCESS_ANY_OWNER` is needed for now. The memory used by ldelf is not supposed to be considered TA private memory. As I said in the second paragraph, we may even have it inaccessible while `ldelf` isn't executing.

Reply:
OK, fine, thanks for the clarification.